synapse/logging/opentracing.py

```diff
@@ -641,57 +641,16 @@ def extract_text_map(carrier):
 # Tracing decorators
 
 
-def trace(func):
+def trace(func=None, operation_name=None):
     """
     Decorator to trace a function.
     Sets the operation name to that of the function's.
```

Comment:
... or that given in the argument.

It might also be worth documenting both ways of using it.